### PR TITLE
kubevirt-builder: Add manual build job

### PR DESCRIPTION
**Pull request by oshoval**
Signed-off-by: Or Shoval <oshoval@redhat.com>

### CODE REVIEW

**Comment by rmohr** [comment]
/hold

I don't think this is an option. We do not publish artifacts from unmerged code in a way that normal users can do it.

**Comment by oshoval** [comment]
> /hold
> 
> I don't think this is an option. We do not publish artifacts from unmerged code in a way that normal users can do it.

removed the publish, and it uses merged code only, isnt it ?

**Comment by rmohr** [comment]
> > /hold
> > I don't think this is an option. We do not publish artifacts from unmerged code in a way that normal users can do it.
> 
> removed the publish, and it uses merged code only, isnt it ?

It looks to me like a presubmit, so it runs on PR, but maybe I miss something. 

If you don't want to publish, please remove the push secrets, then we can have an optional builder-image build-test job.

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml [hunk 1]
```diff
@@ -1899,6 +1899,45 @@ presubmits:
...
+    decorate: true
+    decoration_config:
+      grace_period: 5m0s
+      timeout: 4h0m0s
+    labels:
+      preset-bazel-cache: "true"
+      preset-bazel-unnested: "true"
+      preset-dind-enabled: "true"
+      preset-docker-mirror-proxy: "true"
+      preset-kubevirtci-quay-credential: "true"
```
Comment (rmohr) [inline github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml:1916]: you should be able to remove this.

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml [hunk 2]
```diff
@@ -1899,6 +1899,45 @@ presubmits:
...
+      preset-shared-images: "true"
```
Comment (rmohr) [inline github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml:1917]: you should be able to remove this.

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml [hunk 3]
```diff
@@ -1899,6 +1899,45 @@ presubmits:
...
+    skip_branches:
+    - release-\d+\.\d+
+    spec:
+      containers:
+      - command:
+        - /usr/local/bin/runner.sh
+        - /bin/sh
+        - -ce
+        - |
+          cat "$QUAY_PASSWORD" | docker login --username $(cat "$QUAY_USER") --password-stdin=true quay.io
```
Comment (rmohr) [inline github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml:1930]: This line should not be needed for building only.

**Comment by oshoval** [comment]
addressed comments, thanks

/rehearse

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml [hunk 4]
```diff
@@ -1899,6 +1899,42 @@ presubmits:
...
+    decorate: true
+    decoration_config:
+      grace_period: 5m0s
+      timeout: 4h0m0s
+    labels:
+      preset-bazel-cache: "true"
+      preset-bazel-unnested: "true"
+      preset-dind-enabled: "true"
+      preset-docker-mirror-proxy: "true"
+    max_concurrency: 1
```
Comment (rmohr) [inline github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml:1916]: I don't think we have to limit it. Probably a left-over from the publish job.

**Review by rmohr — CHANGES_REQUESTED** [review]
One last comment. The rest looks good to me.

**Comment by oshoval** [comment]
addressed comments, thanks

File: github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml [hunk 5]
```diff
@@ -1899,6 +1899,41 @@ presubmits:
...
+          make builder-build
+        image: quay.io/kubevirtci/bootstrap:v20220211-d7d6c59
+        name: ""
+        resources:
+          requests:
+            memory: 29Gi
+        securityContext:
+          privileged: true
+      nodeSelector:
+        type: bare-metal-external
```
Comment (rmohr) [inline github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-presubmits.yaml:1936]: Oh, and please remove the nodeSelector. we don't need bare-metal resources for this job.

**Review by rmohr — COMMENTED** [review]
Really last change request. :crossed_fingers:

**Comment by oshoval** [comment]
sure, addressed, thanks ;]

**Comment by oshoval** [comment]
thanks

/rehearse

**Pull request merged**